implementation/src/test/java/io/smallrye/jwt/auth/cdi/ClaimInjectionTest.java

```diff
@@ -21,34 +32,121 @@
 import io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal;
 import io.smallrye.jwt.build.Jwt;
 
-@SuppressWarnings("CdiUnproxyableBeanTypesInspection")
+@SuppressWarnings({
+        "CdiUnproxyableBeanTypesInspection",
+        "OptionalGetWithoutIsPresent"
+})
 public class ClaimInjectionTest {
     @Rule
     public WeldInitiator weld = WeldInitiator.from(
             ClaimInjectionTest.class,
             ClaimsBean.class,
+            ClaimBeanInstance.class,
             RawClaimTypeProducer.class,
+            OptionalClaimTypeProducer.class,
             ClaimValueProducer.class,
+            JsonValueProducer.class,
             CommonJwtProducer.class)
             .addBeans()
-            .activate(RequestScoped.class)
+            .activate(RequestScoped.class, ApplicationScoped.class)
             .inject(this)
             .build();
 
-    @Inject
-    private JsonWebToken jsonWebToken;
     @Inject
     private ClaimsBean claimsBean;
+    @Inject
+    private ClaimBeanInstance claimBeanInstance;
 
     @Test
-    public void inject() {
+    public void injectBoolean() {
         assertTrue(claimsBean.isBooleanClaim());
         assertTrue(claimsBean.getBooleanClaimWrapper());
-        //assertTrue(claimsBean.getBooleanClaimValue().getValue()); // does not unwrap json to wrapper type
+        assertTrue(claimsBean.getBooleanClaimValue().getValue());
+        assertTrue(claimsBean.getBooleanOptional().get());
+        assertTrue(claimsBean.getBooleanClaimValueOptional().getValue().get());
+        // This doesn't work because it causes ambiguous dependencies for JsonValue in JsonValueProducer
+        //assertEquals(claimsBean.getBooleanJson(), JsonValue.TRUE);
```

Comment:
@radcortez Can you clarify please ?

Reply:
Unfortunately, this PR doesn't solve all the cases yet (it does solve most of them). The issue here is with injection of `JsonValue` to represent a `Boolean`. If you look into `JsonValueProducer` we have specialised producers for all Json types, but not for `JsonValue` because it is the super class for all Json types. 

If I add a `JsonValue` producer to solve the `Boolean` case, then we have ambiguous dependencies with all the other types, because `JsonValue` can also be resolver for all the other types.

To fix this case, I need to rewrite `JsonValueProducer` a bit and I wanted to avoid that in this PR.

Reply:
> To fix this case, I need to rewrite JsonValueProducer a bit and I wanted to avoid that in this PR.

+1 - lets iterate